src/Sentry.Unity.Native/SentryNative.cs

```diff
@@ -80,7 +83,26 @@ public static void Configure(SentryUnityOptions options, ISentryUnityInfo sentry
             }
         }
         options.CrashedLastRun = () => crashedLastRun;
+
+        // The backend will only be reinstalled if the native SDK has been initialized successfully.
+        // We're handing this off to be called in `BeforeSceneLoad` instead of `SubsystemRegistration` as it's too soon
+        // and the SignalHandler would still get overwritten.
+        ReinstallBackendFunction = SentryNativeBridge.ReinstallBackend;
     }
 
-    public static void ReinstallBackend() => SentryNativeBridge.ReinstallBackend();
+    [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.BeforeSceneLoad)]
+    public static void ReinstallBackend()
+    {
+        try
+        {
+            // At this point Unity has taken the signal handler and will not invoke our handler. So we register our
+            // backend once more to make sure user-defined data is available in the crash report and the SDK is able
+            // to capture the crash.
+            ReinstallBackendFunction?.Invoke();
+        }
+        catch (EntryPointNotFoundException e)
+        {
+            SentrySdk.CaptureException(new Exception("The SDK failed to reinstall the backend and won't capture native errors.", e));
```

Comment:
I'm not a huge fan of capturing exceptions the SDK throws itself but this is during initialization, making sure that the lack of observability of native crashes is known.

Reply:
I'm not sure this is the best route. What would the user do with this exception?
Also, they'd be paying for these events.

Debug log is the way to go imo

Reply:
How would logging help? I think we have to make sure users can be certain the SDK is functioning correctly.
The scenario I've got in my head and would like to cover:
1. User ships their game but the native SDK fails to reinstall.
2. Crash free rate is 100%
3. Players still report crashes but the user has no idea why Sentry is not reporting any
4. What now? Enable `debug` and ask for logs?

We're doing the same thing to cover the native SDK's `Init` here
https://github.com/getsentry/sentry-unity/blob/8c0967fedc553a2bd0f0ab95ece8b21b6c47e9d2/package-dev/Runtime/SentryInitialization.cs#L77-L92